test/suite/Annotations.test.php

```diff
@@ -585,12 +590,149 @@ protected function testGetAnnotationsFromTrait()
-            self::ANNOTATION_EXCEPTION,
-            "Reading annotations from interface/trait 'TestTrait' is not supported"
-        );
-        Annotations::ofClass('TestTrait', '@note');
+        $annotations = Annotations::ofClass('SimpleTrait', '@note');
+
+        $this->check(count($annotations) === 1, 'SimpleTrait has one note annotation.');
+    }
+
+    protected function testCanGetMethodAnnotationsIncludedFromTrait()
+    {
+        if (version_compare(PHP_VERSION, '5.4.0', '<')) {
+            $this->pass();
+            return;
+        }
+
+        $annotations = Annotations::ofMethod(new \ReflectionClass('SimpleTraitTester'), 'runFromTrait');
+        $this->check(count($annotations) > 0, 'from class reflection and method name');
+
+        $annotations = Annotations::ofMethod(new \ReflectionMethod('SimpleTraitTester', 'runFromTrait'));
+        $this->check(count($annotations) > 0, 'from method reflection');
+
+        $annotations = Annotations::ofMethod(new SimpleTraitTester(), 'runFromTrait');
+        $this->check(count($annotations) > 0, 'from class object and method name');
+
+        $annotations = Annotations::ofMethod('SimpleTraitTester', 'runFromTrait');
+        $this->check(count($annotations) > 0, 'from class name and method name');
+
+        $annotations = Annotations::ofMethod('SimpleTraitTester', 'runFromAnotherTrait');
+        $this->check(count($annotations) > 0, 'from class name and method name for namespaced trait');
+    }
+
+    protected function testHandlesMethodInheritanceWithTraits()
+    {
+        if (version_compare(PHP_VERSION, '5.4.0', '<')) {
+            $this->pass();
+            return;
+        }
+
+        $annotations = Annotations::ofMethod('InheritanceTraitTester', 'baseTraitAndParent');
+        $this->check(count($annotations) === 2, 'baseTraitAndParent inherits parent annotations');
+        $this->check($annotations[0]->note === 'inheritance-base-trait-tester', 'parent annotation first');
+        $this->check($annotations[1]->note === 'inheritance-base-trait', 'trait annotation second');
+
+        $annotations = Annotations::ofMethod('InheritanceTraitTester', 'traitAndParent');
+        $this->check(count($annotations) === 2, 'traitAndParent inherits parent annotations');
+        $this->check($annotations[0]->note === 'inheritance-base-trait-tester', 'parent annotation first');
+        $this->check($annotations[1]->note === 'inheritance-trait', 'trait annotation second');
+
+        $annotations = Annotations::ofMethod('InheritanceTraitTester', 'traitAndChild');
+        $this->check(count($annotations) === 1, 'traitAndChild does not inherit trait');
+        $this->check($annotations[0]->note === 'inheritance-trait-tester', 'child annotation first');
+
+        $annotations = Annotations::ofMethod('InheritanceTraitTester', 'traitAndParentAndChild');
+        $this->check(count($annotations) === 2, 'traitAndParentAndChild does not inherit trait annotation');
```

Comment:
This is correct indeed. Only annotations from class and parent class are merged, but trait annotation is ignored.